### PR TITLE
replace mocha with jest

### PR DESCRIPTION
**Pull request by hiroppy**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### Motivation / Use-Case

https://github.com/webpack/webpack-dev-server/pull/1649#issuecomment-460545627

### Additional Info

Currently, since duplication of ports occurred, the test is executed in series.
I would like to correspond with another PR.
If all CIs pass, remove wip.

### CODE REVIEW

File: test/helpers/run-webpack-dev-server.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ function runWebackDevServer(testArgs, configPath) {
   const args = [webpackDevServerPath, '--config', configPath].concat(testArgs);
 
   return new Promise((resolve, reject) => {
-    const child = spawn('node', args, { cwd, env });
+    const child = execa('node', args, { cwd, env });
```
Comment (hiroppy) [inline test/helpers/run-webpack-dev-server.js]: It is strange to use `spawn`, the following error occurs.

```
 spawn node ENOENT
```

So We will use execa.

**Comment by evilebottnawi** [comment]
Good job, let's wait CI green :+1:

**Comment by evilebottnawi** [comment]
@hiroppy hm, coverage is very decrease, something wrong?

**Comment by hiroppy** [comment]
hmmm.... I don't know that. Current setting is `  "collectCoverageFrom": ["(lib|bin)/**/*.js"],`.

**Comment by hiroppy** [comment]
Test failed on appveyor.(Environment: nodejs_version=8, webpack_version=latest)

```
FAIL test/Routes.test.js
  ● Routes › headers as an array › GET request with headers as an array
    expected "X-Bar" of "key1=value1, key2=value2", got "key1=value1,key2=value2"
      at Test.Object.<anonymous>.Test._assertHeader (node_modules/supertest/lib/test.js:249:12)
      at Test.Object.<anonymous>.Test._assertFunction (node_modules/supertest/lib/test.js:283:11)
      at Test.Object.<anonymous>.Test.assert (node_modules/supertest/lib/test.js:173:18)
      at Server.assert (node_modules/supertest/lib/test.js:131:12)
```

**Comment by hiroppy** [comment]
@evilebottnawi Fixed. https://codecov.io/gh/webpack/webpack-dev-server/commit/dfe5bade21a4bd9f02b40cb497cefaee868adae0

**Comment by hiroppy** [comment]
The following error occurs only at Node 8 🤔(it also happened on my mac)

```
 expected "X-Bar" of "key1=value1, key2=value2", got "key1=value1,key2=value2"
```

**Comment by hiroppy** [comment]
@evilebottnawi Since I do not know the cause, I temporarily verified it.
I think that it is not good, but I think that the influence range is not so big either.
https://github.com/webpack/webpack-dev-server/pull/1650/commits/29b320d7141725138816367f1cb667d4d531d1e2

File: test/Routes.test.js [hunk 2]
```diff
@@ -101,12 +101,16 @@ describe('Routes', () => {
...
     });
 
-    after(helper.close);
+    afterAll(helper.close);
 
     it('GET request with headers as an array', (done) => {
+      const expected =
+        process.version.split('.')[0] === 'v8'
+          ? 'key1=value1,key2=value2'
+          : 'key1=value1, key2=value2';
```
Comment (evilebottnawi) [inline test/Routes.test.js]: Maybe just remove spaces from expected? I am afraid another version can also have same problem?

Comment (hiroppy) [reply test/Routes.test.js]: | version | output |space|
| :--- | :--- |:--- |
| 6| `key1=value1, key2=value2` |y|
| 7 | `key1=value1,key2=value2` |n| 
| 8| `key1=value1,key2=value2` |n|
| 9 | `key1=value1,key2=value2` |n| 
| 10| `key1=value1, key2=value2` |y|
| 11 | `key1=value1, key2=value2` |y|

6, 10, 11 have the spaces but 7, 8, 9 don't have.

So, I change the code to `['v7', 'v8', 'v9'].includes`.

Comment (evilebottnawi) [reply test/Routes.test.js]: @hiroppy Can you send this change in commit?

Comment (hiroppy) [reply test/Routes.test.js]: added to https://github.com/webpack/webpack-dev-server/pull/1650/files#diff-f1603d639761fe762d975b5ce05e20b1R107

Comment (hiroppy) [reply test/Routes.test.js]: Separated this commit(https://github.com/webpack/webpack-dev-server/pull/1650/commits/10e4bc27c96b6c70b8020cf2e034796a5b552fc0) /cc @evilebottnawi

**Comment by evilebottnawi** [comment]
@hiroppy Good job, let's wait CI green

**Comment by hiroppy** [comment]
@evilebottnawi Thanks for reviewing! CI is green:)

**Comment by evilebottnawi** [comment]
@hiroppy let's improve CI testing 

https://github.com/webpack/webpack-dev-server/blob/master/.travis.yml (need add 8 and 11)

https://github.com/webpack/webpack-dev-server/blob/master/appveyor.yml (need add node 11)

Maybe refactor and improve config (add cache)

Can you take care?

**Comment by hiroppy** [comment]
Yep, I'll work on that:)

**Pull request merged**